src/main/java/datawave/query/util/FrequencyFamilyCounter.java

```diff
@@ -0,0 +1,127 @@
+package datawave.query.util;
+
+import org.apache.accumulo.core.data.Value;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class FrequencyFamilyCounter {
+    
+    private long total = 0L;
+    private HashMap<String,Integer> dateToFrequencyValueMap = new HashMap<>();
+    private static final int SIMPLEDATE_LENGTH = 8;
+    private DateFrequencyValue serializer = new DateFrequencyValue();
+    
+    // private static Pattern SimpleDatePattern = Pattern.compile("^(19|20)\\d\\d[- /.] (0[1-9]|1[012])[- /.] (0[1-9]|[12][0-9]|3[01])$");
+    
+    private static final Logger log = LoggerFactory.getLogger(FrequencyFamilyCounter.class);
+    
+    public FrequencyFamilyCounter() {
+        
+    }
+    
+    public void initialize(Value value) {
+        deserializeCompressedValue(value);
+    }
+    
+    public void clear() {
+        dateToFrequencyValueMap.clear();
+    }
+    
+    public HashMap<String,Integer> getDateToFrequencyValueMap() {
+        return dateToFrequencyValueMap;
+    }
+    
+    /**
+     * Takes the value of a compressed f column record and creates the qualifierToFrequencyValueMap so that other records that are not yet aggregated can be
+     * added to the compressed record. After those records are aggregated they are discarded.
+     *
+     * @param oldValue
+     */
+    public void deserializeCompressedValue(Value oldValue) {
+        
+        HashMap<String,Integer> uncompressedValueMap = serializer.deserialize(oldValue);
+        dateToFrequencyValueMap.clear();
+        dateToFrequencyValueMap.putAll(uncompressedValueMap);
```

Comment:
You apparently just love making copies of things.  Why not assign uncompressedValueMap to dateToFrequencyValueMap?